### DISCUSSION TITLE
Unwanted margin at headers top

### DISCUSSION THREAD

**Discussion by idenisovs** [gotenberg/gotenberg#343] [Q&A] [answered]
Hi,

Is it possible to remove unwanted margin at the top of header? 

At the moment it look like in the screenshot.

In `header.html`:

```css
html, body {
    margin: 0;
    padding: 0;
    font-size: 13pt;
}
```

In PHP:

```php
$request = new HTMLRequest($contentHtml);
$request->setPaperSize(Request::A4);
$request->setHeader($headerHtml);
$request->setFooter($footerHtml);
```

Gotenber version **6.x**

`setMargins` affects just `content.html`, `header.html` still has the margin.

```php
$request->setMargins(Request::NO_MARGINS);
```

![Screenshot from 2021-09-09 13-26-25](https://user-images.githubusercontent.com/16684894/132670239-539a9f37-e009-47e7-b077-8191001e2a24.png)

**Comment by gulien** [comment, marked as answer]
Hello @idenisovs,

Could you try this solution https://github.com/puppeteer/puppeteer/issues/4132#issuecomment-475110167? If it works, I’ll update the documentation 😄

**Reply by idenisovs** [reply]
Hi, @gulien! Thanks for you answer! Just tried to apply `padding: 0 !important`, unfortunately, that solution is not working.

**Reply by gulien** [reply]
Hmm, did you add the style directly in the `header.html` file?

**Reply by idenisovs** [reply]
@gulien Yes, in header.html. 

I came to different solution, by setting margin bottom negative value, like that: `margin-bottom: -15px;`, but anyway, that is strange.

**Reply by gulien** [reply]
Seems weird but hey, if it works 😅

**Reply by neongrau** [reply]
Started with trying to figure out the spacing above and below too, then went on to figure out the scaling issue.
As i have some landscape and some portrait PDFs to print i tried to find a general solution and it seems this works regardless of orientation and page size/width.

> Note the `#header` and `#footer` ids are internally used by chromium.
> If you already use these ids in your HTML and CSS for PDF generating you are in for a bad ride. 
> So get rid of them if you have and add this snippet to your header and footer HTML.

```css
#header, #footer {
  padding: 0 !important;
  transform: scale(0.75) !important;
  transform-origin: top left;
  height: 100% !important;
  position: fixed;
}
```

While the transform scale alone would just make the width smaller, setting `height` and `position: fixed` as well will help render it at the same original width (weird actually, but not complaining) without having to set the width explicitly. At least for me. Maybe it helps the next one running into this issue.

FYI: this all goes back to a chromium bug known since March 2018 but hasn't been fixed until today.
https://bugs.chromium.org/p/chromium/issues/detail?id=818496
